src/main/java/seedu/address/logic/parser/ListClientCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.logic.parser;
+
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRODUCT;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CLIENTS;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.ListClientCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.product.Product;
+
+/**
+ * Parses input arguments and creates a new ListClientCommand object
+ */
+public class ListClientCommandParser {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ViewMeetingCommand
+     * and returns a ViewMeetingCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListClientCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_PRODUCT);
+
+        if (!arePrefixesPresent(argumentMultimap, PREFIX_PRODUCT)) {
+            return new ListClientCommand(PREDICATE_SHOW_ALL_CLIENTS);
+        } else {
+            Product product = ParserUtil.parseProduct(argumentMultimap.getValue(PREFIX_PRODUCT).get());
+            return new ListClientCommand(client -> client.hasProduct(product));
+        }
```

Comment:
It might be better to pass the product straight to the ListClientCommand and then form the predicate inside the command itself. IMO, the parser classes should be concerned with parsing only (i.e. turning the argument strings into the command).